common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -520,14 +530,64 @@ private void processWebsiteRequest(@NonNull final WebView view, @NonNull final S
                 }
             }
 
-            // Otherwise, launch in Browser.
+            loadDeviceCaUrl(url, view);
+        } else {
+            Logger.info(methodTag, "Not a device CA request. Redirecting to browser.");
             openLinkInBrowser(url);
+            returnResult(RawAuthorizationResult.ResultCode.CANCELLED);
+        }
+    }
+
+    private boolean isDeviceCaRequest(@NonNull final String url) {
+        return url.contains(AuthenticationConstants.Broker.BROWSER_DEVICE_CA_URL_QUERY_STRING_PARAMETER);
+    }
+
+    private boolean shouldLaunchCompanyPortal() {
+        PackageHelper packageHelper = new PackageHelper(getActivity().getPackageManager());
+
+        return packageHelper.isPackageInstalledAndEnabled(IPPHONE_APP_PACKAGE_NAME)
+                && IPPHONE_APP_SIGNATURE.equals(packageHelper.getSha1SignatureForPackage(IPPHONE_APP_PACKAGE_NAME))
+                && packageHelper.isPackageInstalledAndEnabled(COMPANY_PORTAL_APP_PACKAGE_NAME);
+    }
+
+    @VisibleForTesting
+    protected void loadDeviceCaUrl(@NonNull final String originalUrl, @NonNull final WebView view) {
+        final String methodTag = TAG + ":loadDeviceCaUrl";
+        final SpanContext spanContext = getActivity() instanceof AuthorizationActivity ? ((AuthorizationActivity) getActivity()).getSpanContext() : null;
+        final Span span = spanContext != null ?
```

Comment:
nit: do we need a new span? can we use the parent acquireToken span?

Reply:
Yes I need a span to easily track this flow as it is only being enabled for some tenants. If I use the parent span, it could get sampled a lot more than I need.